iroh-bytes/src/store/flat.rs

```diff
@@ -1201,21 +1251,82 @@ impl Store {
         root.join("meta")
     }
 
+    /// Path to the redb file where is stored.
+    pub(crate) fn db_path(root: &Path) -> PathBuf {
+        Self::meta_path(root).join("db.v0")
+    }
+
     /// scan a directory for data
     pub(crate) fn load_sync(path: &Path) -> anyhow::Result<Self> {
         tracing::info!("loading database from {}", path.display(),);
         let complete_path = Self::complete_path(path);
         let partial_path = Self::partial_path(path);
         let meta_path = Self::meta_path(path);
+        let db_path = Self::db_path(path);
+
+        let needs_migration = !db_path.exists();
```

Comment:
If the migration is canceled or errors, the db file will exists, but the migration won't be done. Maybe check for all tables to exist to determine if the migration is needed, and only create the tables in the migration inside the transaction there.

Reply:
I guess I could also use the tags.meta as an indicator. That is gone afterwards.

Reply:
ended up just using an entirely new directory so the distinction is clear.